Yoast/wordpress-seo
# Make the help center responsive

## Summary

Responsive layout for the Help Center and video. This PR focuses on the functional implementation, trying to make it work in a consistent fashion across different browsers 😱 

Maybe the Help Center could benefit from a bit of art direction and details refinements. I'd say they would be very welcome in future iterations but I'd propose to focus on the functionality, for now.

Making the layout responsive must take into account 3 different main conditions (not to mention browsers inconsistencies):

- the WordPress admin menu can be expanded or collapsed
- posts, pages, media, and CPTs edit screen can be set to a 2 columns or 1 column layout (from the Screen Options setting)
- Premium has just one text box on the right of the video

Trying to recap how it works:
- **requirement: the video must have a 16:9 ratio**
- the video has an initial width of 560px (this can be changed in the future)
- about the two text boxes:
  - when they're on the right of the video, they're one on top of the other
  - when they go below the video, they have the same width of the video
- on `post.php` the page layout can be in 2 columns or 1 column
  - when 1 column: the Help Center looks identical and behave the same way the one on the Setting pages
  - when in 2 columns: the layout transformations while shrinking the window must happen earlier because the available width is smaller
- under a 782px viewport width all the content gets linearized and the video is full-width

What's inside the diff:

- in `help-center.scss` replaces all the spaces used for indentation with tabs
- makes the layout and video responsive (see inline comments)
- adds comments with implementation details
- removes `overflow: auto;` from `#wpseo_meta .inside` (please double check this)
- fixes headings hierarchy
- uses a CSS class to toggle the Help Center visibility
- renames a few CSS classes to avoid conflicts with WP CSS classes
- removes the full-width purple background hack


## Test instructions

To test, build the JS and CSS. Please remember to test all the places where the Help Center is used: Post, Page, Custom Post Types, Taxonomies, and Media should be basically the same. Settings pages work a bit differently.
I've tested with Chrome 55, Firefox 51, Safari 10, IE11 and Edge. More testing greatly recommended, especially on real mobile devices.
Suggestions for testing:
- go in any edit post screen
- set it to a 1 column layout from the WordPress Screen Options
- open a Yoast SEO settings page in a new tab
- the Help Center in the post screen and the one in the Settings page should have the same, identical, size and behavior
- change viewport width, expand/collapse the admin menu, etc. do all the testing
- then, the post screen needs to be tested in the two columns layout, where all the transformations happen earlier

Gallery of images here: https://cloudup.com/cDjLXttdQ9R

Note: on `post.php` WordPress uses a CSS rule that can break the responsiveness of the whole page under certain conditions.
If you see issues with the Help Center not being responsive on `post.php`, please check first this in your browser DOM inspector and disable it:

```
#poststuff {
    ...
    min-width: 763px; <--
}
```
(gets reset under 850px)

Fixes #6489
